lib/Graph/Graph.cpp

```diff
@@ -682,6 +682,35 @@ Node *Function::createChannelShuffle(llvm::StringRef name, NodeValue input,
   return createReshape(name.str() + ".reshape2", T, inDims);
 }
 
+Node *Function::createSqueeze(llvm::StringRef name, NodeValue input,
+                              llvm::ArrayRef<size_t> axes) {
+  auto originalSize = axes.size();
+  GLOW_ASSERT(originalSize > 0 && "Parameter `axes` must be provided.");
+
+  ShapeVector shapeAxes(axes.begin(), axes.end());
+  std::sort(shapeAxes.begin(), shapeAxes.end());
+  shapeAxes.erase(std::unique(shapeAxes.begin(), shapeAxes.end()),
+                  shapeAxes.end());
+  GLOW_ASSERT(shapeAxes.front() >= 0 && "Dimension ids must be non-negative.");
+
+  auto inDims = input.dims();
+  GLOW_ASSERT(
+      shapeAxes.size() < inDims.size() &&
```

Comment:
Let's instead check stronger requirement: `shapeAxes.back() < inDims.size()`. It will automatically imply that `shapeAxes.size() < inDims.size()`, and we need to check it anyway, to make sure that no `shapeAxes` element is greater than number of dimensions.